clip2onnx/model_card.py

```diff
@@ -0,0 +1,165 @@
+import inspect
+import logging
+import sys
+import warnings
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import clip
+import open_clip
+import torch
+from PIL import Image
+
+warnings.filterwarnings("ignore", category=UserWarning)
+
+logging.basicConfig(
+    level=logging.INFO, stream=sys.stdout, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+@dataclass
+class ModelCard:
+    """
+    Open-CLIP Benchmarks
+    > https://github.com/mlfoundations/open_clip/blob/main/docs/openclip_results.csv
+
+    Open-CLIP Model Profile
+    > https://github.com/mlfoundations/open_clip/blob/main/docs/model_profile.csv
+
+    """
+
+    model_name: str
+    """
+    regular model name
+    """
+
+    tag: str = ""
+    """
+    pretrained
+    """
+
+    onnx_visual: str = ""
+    onnx_textual: str = ""
+    """
+    Merged ONNX Model Name.
+    If you decide to export the model on the huggingface, 
+    please customize the onnx_visual and onnx_textual fields.
+    """
+
+    onnx_visual_path: Path = field(default_factory=Path)
+    onnx_textual_path: Path = field(default_factory=Path)
+    model_dir: Path = Path("model")
+    """
+    Default export path:
+      - visual: [project_dir]/model/*[self.model_name]/[VERSION]/[self.onnx_visual]
+      - textual: [project_dir]/model/*[self.model_name]/[VERSION]/[self.onnx_textual]
+    """
+
+    DEFAULT_TEXTUAL_FIELDS = {
+        "export_params": True,
+        "input_names": ["TEXT"],
+        "output_names": ["TEXT_EMBEDDING"],
+        "dynamic_axes": {"TEXT": {0: "text_batch_size"}, "TEXT_EMBEDDING": {0: "text_batch_size"}},
+    }
+    """
+    Template parameter of the textual-part appended to `torch.onnx.export`
+    """
+
+    DEFAULT_VISUAL_FIELDS = {
+        "export_params": True,
+        "input_names": ["IMAGE"],
+        "output_names": ["IMAGE_EMBEDDING"],
+        "dynamic_axes": {
+            "IMAGE": {0: "image_batch_size"},
+            "IMAGE_EMBEDDING": {0: "image_batch_size"},
+        },
+    }
+    """
+    Template parameter of the visual-part appended to `torch.onnx.export`
+    """
+
+    def __post_init__(self):
+        _prefix = self.model_name
```

Comment:
No need to name variables with a leading underscore if you're inside of a method - they're not accessible from the outside in any case